### PR TITLE
Car filter

### PR DESCRIPTION
**Pull request by sebastianottow**
Add a toggle up top to toggle between views of only showing favorite cars vs all cars.
Also added that marking a car with favorite will move it up in the list.

### CODE REVIEW

File: src/components/CarFilter/CarFilter.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { CheckBox, CheckBoxLabel, CheckBoxWrapper } from './StyledCarFilter';
+import StyledDiv from '../../../styles/StyledDiv';
+import useStore from '../../lib/hooks/useStore';
+
+export default function CarFilter({ checkState }) {
+	const toggleFilter = useStore(state => state.toggleFilter);
+
+	return (
+		<StyledDiv margin="2vh 0 0 0 ">
```
Comment (anitaschiller) [inline src/components/CarFilter/CarFilter.js:11]: Cool 😎

File: src/components/CarFilter/StyledCarFilter.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			display: block;
+			border-radius: 50%;
+			width: 18px;
+			height: 18px;
+			margin-left: 21px;
+			transition: 0.2s;
+		}
+	}
+`;
+export { CheckBox };
```
Comment (anitaschiller) [inline src/components/CarFilter/StyledCarFilter.js]: Exporting several styled-components from one file might be confusing later. I'd recommend creating a single file for each styled-component and only export ONE per file.

File: src/components/Card/Card.js [hunk 3]
```diff
@@ -6,15 +6,17 @@ import StyledButton from '../../../styles/StyledButton';
...
 
-export default function Card() {
-	const myVehicles = useStore(state => state.myVehicles);
+export default function Card({ data }) {
 	const toggleDeleteState = useStore(state => state.toggleDeleteState);
 	const setID = useStore(state => state.setID);
 	const setVehicleName = useStore(state => state.setVehicleName);
+	const sortedVehicles = data.sort((a, b) => b.favorite - a.favorite);
+
+	//const trueFirst = arr.sort((a, b) => Number(b.bool) - Number(a.bool));
```
Comment (anitaschiller) [inline src/components/Card/Card.js]: I think this can be deleted?

File: src/components/Card/Card.js [hunk 4]
```diff
@@ -6,15 +6,17 @@ import StyledButton from '../../../styles/StyledButton';
 import StyledFlex from '../../../styles/StyledFlex';
 import StyledCard from './StyledCard';
 
-export default function Card() {
-	const myVehicles = useStore(state => state.myVehicles);
+export default function Card({ data }) {
 	const toggleDeleteState = useStore(state => state.toggleDeleteState);
 	const setID = useStore(state => state.setID);
 	const setVehicleName = useStore(state => state.setVehicleName);
+	const sortedVehicles = data.sort((a, b) => b.favorite - a.favorite);
```
Comment (anitaschiller) [inline src/components/Card/Card.js:13]: I like how you sort your vehicles. I would have probably used two filter-functions but your solution is shorter and quite smart ;)

File: src/components/CardGrid/CardGrid.js [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
+
 	return (
-		<StyledFlex margin="4vh 10px 12vh 10px" gap="40px" alignItems="center">
-			<Card />
-		</StyledFlex>
+		<>
+			<CarFilter checkState={filterState} />
+			<StyledFlex margin="4vh 10px 12vh 10px" gap="40px" alignItems="center">
+				{filterState && <Card data={myFavoriteVehicles} />}
+				{!filterState && <Card data={myVehicles} />}
```
Comment (anitaschiller) [inline src/components/CardGrid/CardGrid.js]: An alternative could be: 
```suggestion
				{filterState ? <Card data={myFavoriteVehicles} /> : <Card data={myVehicles} />}
```

**Review by anitaschiller — CHANGES_REQUESTED** [review]
Hi Sebastian, nice job! 
If you shift your styled-components in separate files you are ready to go from my side 😉.

File: src/components/Card/Card.js (hunk 3 above)
Comment (sebastianottow) [reply src/components/Card/Card.js]: absolutely

File: src/components/Card/Card.js (hunk 4 above)
Comment (sebastianottow) [reply src/components/Card/Card.js:13]: Thanks :) Took me a while to figure this out :D

File: src/components/CarFilter/StyledCarFilter.js (hunk 2 above)
Comment (sebastianottow) [reply src/components/CarFilter/StyledCarFilter.js]: Done.

Comment (sebastianottow) [reply src/components/CarFilter/StyledCarFilter.js]: Done :)

File: src/components/CardGrid/CardGrid.js (hunk 5 above)
Comment (sebastianottow) [reply src/components/CardGrid/CardGrid.js]: Yep, implemented since it is shorter :) Thx

**Pull request merged**